lib/ipmi_user.c

```diff
@@ -434,7 +434,7 @@ print_user_usage(void)
 	lprintf(LOG_NOTICE,
 "               set name     <user id> <username>");
 	lprintf(LOG_NOTICE,
-"               set password <user id> [<password> <16|20>]");
+"               set password <user id> [<password> [<16|20>]]");
```

Comment:
Don't you want to also change the code so that if the actual password string length is longer than 16, it would automatically set length to 20 ? So far, as I can see, it assumes 16 for both command line and stdin passwords, but always uses 20 when checking for the length limit. Doesn't look correct to me. If you're not willing to fix that, it's ok, I can do it myself later.

Reply:
Thanks for looking it over. I agree the length handling is a bit odd but I wasn't sure that detecting the size was necessarily "correct" or consistent behaviour.  I'd be happy to implement it at your suggestion.

At the moment it will return an error when the password is over 20 characters, but that is at odds with the truncation behaviour for a password of length 16.

I'd suggest that the code should behave consistently: if the password length exceeds an _explicit_ length, that seems like an error condition.  But if no length is specified, it would be OK to automatically detect (and raise an error if the max length is exceeded).

(But isn't the whole thing complicated by the fact that 20-character passwords aren't accepted over some communications channels?  Or did I misremember that?)

Reply:
I suppose the BMC will return an error if a >16 character password is attempted when IPMI 2.0 isn't supported. Hence, I think the behavior you propose would be just fine:

> I'd suggest that the code should behave consistently: if the password length exceeds an explicit length, that seems like an error condition. But if no length is specified, it would be OK to automatically detect (and raise an error if the max length is exceeded).